extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOIDCProvidersRecorder.java

```diff
@@ -0,0 +1,111 @@
+package io.quarkus.oidc.runtime.providers;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import io.quarkus.oidc.OidcTenantConfig;
+import io.quarkus.oidc.runtime.OidcConfig;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class KnownOIDCProvidersRecorder {
+
+    final OidcConfig config;
+
+    public KnownOIDCProvidersRecorder(OidcConfig config) {
+        this.config = config;
+    }
+
+    public Supplier<Optional<OidcTenantConfig>> github() {
+        return new Supplier<Optional<OidcTenantConfig>>() {
+            @Override
+            public Optional<OidcTenantConfig> get() {
+                if (config.provider.github.isEmpty()) {
+                    return Optional.empty();
+                }
+                OidcTenantConfig ret = new OidcTenantConfig();
+                ret.clientId = Optional.of(config.provider.github.get().clientId);
+                ret.credentials.secret = Optional.of(config.provider.github.get().secret);
+                ret.authServerUrl = Optional.of("https://github.com/login/oauth");
+                //TODO: do we want to hard code this?
+                ret.applicationType = OidcTenantConfig.ApplicationType.HYBRID;
+                ret.discoveryEnabled = false;
+                ret.authorizationPath = Optional.of("authorize");
+                ret.tokenPath = Optional.of("access_token");
+                ret.userInfoPath = Optional.of("https://api.github.com/user");
+                ret.authentication.scopes = Optional.of(config.provider.github.get().scopes);
+                ret.authentication.userInfoRequired = true;
+                ret.authentication.setIdTokenRequired(false);
+                ret.authentication.setRedirectPath("/Login/githubLoginSuccess");
```

Comment:
I think those links are too specific to my app. But in practice:
- google/apple/ms works on a single endpoint without requiring `UserInfo`: everything is in the ID Token
- github must be handled specially because `UserInfo` contains the user ID, name and user name, but we have to hit a second endpoint `/user/emails` to get the user's email.
- facebook must be handled specially because `UserInfo` contains the user ID, name and email

Perhaps a better endpoint default would be: `/{provider}-oidc-login-success`?

Reply:
Naturally, a further layer of abstraction which knows how to fetch the damn fields from all providers and present them to the Quarkus user would be even better: then I only need a single success endpoint :)
But that can wait for another issue.